decidim-core/app/jobs/decidim/metric_job.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Decidim
+  class MetricJob < ApplicationJob
+    queue_as :metrics
+
+    def perform(manager_class, organization_id, day = nil)
+      metric = manager_class.constantize.for(day)
+      metric.with_context(Decidim::Organization.find_by(id: organization_id))
+      metric.query
+      metric.registry!
```

Comment:
I find the api for the manager class a bit complicated. What about something like this:

```ruby
organization = Decidim::Organization.find_by(id: organization_id)
metric = manager_class.constantize.for(day, organization)
metric.record
```

And the `record` method handles everything, having to call `with_context` and `query` is not very intuitive.

Reply:
Although I'd review my first comment: https://github.com/decidim/decidim/pull/3603/files#r208531369

Why can't we use this instead of having the manager class?

Also, we should add a check at this job to not store a metric if it already exists for this day (and probably add an index to the database)